llvm/test/Transforms/InstCombine/or.ll

```diff
@@ -1566,14 +1566,14 @@ define i32 @mul_no_common_bits_uses(i32 %p1, i32 %p2) {
   ret i32 %r
 }
 
-; negative test - probably not good to create an extra mul
+; TODO: is it good to create an extra mul?
 
 define i32 @mul_no_common_bits_const_op_uses(i32 %p) {
 ; CHECK-LABEL: @mul_no_common_bits_const_op_uses(
 ; CHECK-NEXT:    [[X:%.*]] = and i32 [[P:%.*]], 7
 ; CHECK-NEXT:    [[M:%.*]] = mul nuw nsw i32 [[X]], 24
 ; CHECK-NEXT:    call void @use(i32 [[M]])
-; CHECK-NEXT:    [[R:%.*]] = or disjoint i32 [[M]], [[X]]
+; CHECK-NEXT:    [[R:%.*]] = mul nuw nsw i32 [[X]], 25
```

Comment:
We need the `one-use` check.

Reply:
> We need the `one-use` check.
> 
> 

Why do we have it for Or but not Add?

Reply:
Should probably be present for add as well. We don't have a hard rule on this, but replacing an add with a multiply does not seem profitable to me.

Reply:
I think I agree we should have the one use check for both add and or.  I'm entirely willing to let this go in so that we treat the disjoint or as an add, then fix the one use issue separately though.  They seem like orthogonal issues.

Reply:
I did some investigation of how to add the one use check. It's a little complicated.  The optimization can produce a Mul that can be transformed into a Shl. In that case the one use check wouldn't need to apply. The optimization can also turn a Shl into a Mul that would need the one use check to apply.